otter/util/deferredutils.py

```diff
@@ -114,6 +114,24 @@ def retry_and_timeout(do_work, timeout, can_retry=None, next_interval=None,
     Retry a function until the function succeeds or timeout has been reached.
     This is just a composition of :func:`timeout_deferred` and :func:`retry`
     for convenience.  Please see their respective arguments.
+
+    :param callable do_work: Takes no arguments.  Implements the work you want
+        to perform on a periodic basis.
+    :param number timeout: The number of seconds beyond which is considered a
+        timeout condition.
+    :param callable can_retry: Takes a Twisted Failure instance as a
+        parameter, and decides whether or not the work should be retried (True
```

Comment:
Non-blocking: could you please mention: `and returns bool that tells whether the work should be retried`? The emphasis is on `returns bool` :)